integrated_channels/api/v1/cornerstone/views.py

```diff
@@ -1,17 +1,104 @@
 """
 Viewsets for integrated_channels/v1/cornerstone/
 """
+from logging import getLogger
+
+from django.contrib import auth
+from django.db import IntegrityError, transaction
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+
 from rest_framework import permissions, viewsets
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.response import Response
+from rest_framework.status import HTTP_200_OK, HTTP_400_BAD_REQUEST
+from rest_framework.views import APIView
 
+from enterprise.api.throttles import ServiceUserThrottle
+from enterprise.utils import get_enterprise_customer_or_404, localized_utcnow
 from integrated_channels.api.v1.mixins import PermissionRequiredForIntegratedChannelMixin
-from integrated_channels.cornerstone.models import CornerstoneEnterpriseCustomerConfiguration
+from integrated_channels.cornerstone.models import (
+    CornerstoneEnterpriseCustomerConfiguration,
+    CornerstoneLearnerDataTransmissionAudit
+)
+from integrated_channels.cornerstone.utils import create_cornerstone_learner_data
 
 from .serializers import CornerstoneConfigSerializer
 
+LOGGER = getLogger(__name__)
+User = auth.get_user_model()
+
 
 class CornerstoneConfigurationViewSet(PermissionRequiredForIntegratedChannelMixin, viewsets.ModelViewSet):
     serializer_class = CornerstoneConfigSerializer
     permission_classes = (permissions.IsAuthenticated,)
     permission_required = 'enterprise.can_access_admin_dashboard'
 
     configuration_model = CornerstoneEnterpriseCustomerConfiguration
+
+
+class CornerstoneLearnerInformationView(APIView):
+    permission_classes = (permissions.IsAuthenticated,)
+    authentication_classes = (JwtAuthentication, SessionAuthentication,)
+    throttle_classes = (ServiceUserThrottle,)
+
+    def post(self, request):
+        """
+            POST integrated_channels/api/v1/cornerstone/save-learner-information
+            Requires a JSON object of the following format:
+                {
+                    "courseKey": "course-v1:edX+DemoX+Demo_Course",
```

Comment:
nit: we are passing the course_key from frontend not the course run key.

Reply:
done